src/NUnitFramework/tests/Constraints/AnyOfConstraintTests.cs

```diff
@@ -0,0 +1,59 @@
+// ***********************************************************************
+// Copyright (c) 2009 Charlie Poole, Rob Prouse
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using System;
+namespace NUnit.Framework.Constraints
+{
+    [TestFixture]
+    public class AnyOfConstraintTests
```

Comment:
These are not just Constraint tests. They are testing the constraint as used __through__ the `Is.AnyOf` syntax. That's not a bad thing to do, but we generally test constraints and syntax elements separately. I would not get rid of these, but I'd like to add the separate tests.

In particular, testing a constraint separately allows us to easily write tests of the description and the error message. In many cases (not this one, of course) we have alternative syntax that can lead to the same constraint. So to test the syntax, we can simply verify that it produces the proper constraints, which we do by examining the string representation of the generated constraint expression.

Reply:
Syntax folder.